mmv1/third_party/terraform/resources/resource_apigee_keystores_aliases_key_cert_file.go

```diff
@@ -0,0 +1,674 @@
+package google
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"mime/multipart"
+	"reflect"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceApigeeKeystoresAliasesKeyCertFile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceApigeeKeystoresAliasesKeyCertFileCreate,
+		Read:   resourceApigeeKeystoresAliasesKeyCertFileRead,
+		Update: resourceApigeeKeystoresAliasesKeyCertFileUpdate,
+		Delete: resourceApigeeKeystoresAliasesKeyCertFileDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceApigeeKeystoresAliasesKeyCertFileImport,
+		},
+
+		CustomizeDiff: customdiff.All(
+			/*
+				If cert is changed then an update is expected, so we tell Terraform core to expect update on certs_info
+			*/
+
+			customdiff.ComputedIf("certs_info", func(_ context.Context, diff *schema.ResourceDiff, v interface{}) bool {
+				return diff.HasChange("cert")
+			}),
+		),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+			Read:   schema.DefaultTimeout(20 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(20 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"alias": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: `Alias Name`,
+			},
+			"cert": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Cert content`,
+			},
+			"environment": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: `Environment associated with the alias`,
+			},
+			"keystore": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: `Keystore Name`,
+			},
+			"org_id": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: `Organization ID associated with the alias`,
+			},
+			"certs_info": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
```

Comment:
@hao-nan-li  This `certInfo` is a nested Object that contains all the info of the uploaded cert, this is parsed from the uploaded cert by API side and then returned, so we set it as computed. This can be used to detect drift, let's say if cert is being modified outside the Terraform the READ operation will detect drift